File: jobs/scripts/transaction-file.py
import os
import mysql.connector
import json

mydb = mysql.connector.connect(
  host=os.getenv('DB_URL'),
  user=os.getenv('DB_USER'),
  password=os.getenv('DB_PASSWORD'),
  database=os.getenv('DB_NAME')
)

transactions_file = open('jobs/batches/transactions.json')

transactions = json.load(transactions_file)

mycursor = mydb.cursor()
number_transactions = len(transactions)
number_processed = 0
try:
  for transaction in transactions:
      sql = f"UPDATE clients SET balance = balance {'+' if transaction['OpType']=='IN' else '-'} {transaction['Amount']} WHERE code = {transaction['Code']}"

      mycursor.execute(sql)

      mydb.commit()

      number_processed += 1
except Exception as Error:
  print(f"{Error=}")

print('::', (number_processed / number_transactions) * 100, '%', 'Completed')
  

File: jobs/scripts/transaction-file.go
package main

import (
	"database/sql"
	"encoding/json"
	"fmt"
	"log"
	"os"

	"github.com/bitfield/script"
	_ "github.com/go-sql-driver/mysql"
	"github.com/joho/godotenv"
)

type Transaction struct {
	Code   uint
	Amount float64
	OpType string
}

func main() {
	godotenv.Load(".env")
	var transactions []Transaction

	transactionsFromJson, errIo := script.File("jobs/batches/transactions.json").String()
	if errIo != nil {
		log.Fatal(errIo)
	}

	json.Unmarshal([]byte(transactionsFromJson), &transactions)

	db, err := sql.Open("mysql", fmt.Sprintf(
		"%s:%s@tcp(%s:%s)/%s?charset=utf8mb4&parseTime=True&loc=Local",
		os.Getenv("DB_USER"),
		os.Getenv("DB_PASSWORD"),
		os.Getenv("DB_URL"),
		os.Getenv("DB_PORT"),
		os.Getenv("DB_NAME"),
	))

	defer db.Close()

	if err != nil {
		log.Fatal(err)
	}

	var op string

	for _, transaction := range transactions {
		if transaction.OpType == "IN" {
			op = "+"
		} else {
			op = "-"
		}
		updateQuery := fmt.Sprintf("UPDATE clients SET balance = balance %s %f WHERE code = %d", op, transaction.Amount, transaction.Code)

		db.Query(updateQuery)
	}
}
